examples/gallery/plot/points-categorial.py

```diff
@@ -0,0 +1,48 @@
+"""
+Color points by categories
+---------------------------
+The :meth:`pygmt.Figure.plot` method can be used to plot symbols which are
+color-coded by categories.
+"""
+
+import numpy as np
+import pandas as pd
+import pygmt
+
+# Load sample iris data, and convert 'species' column to categorical dtype
+df = pd.read_csv("https://github.com/mwaskom/seaborn-data/raw/master/iris.csv")
+df["species"] = df.species.astype(dtype="category")
+
+# Use pygmt.info to get region bounds (xmin, xmax, ymin, ymax)
+# The below example will return a numpy array like [2.  4.4 4.3 7.9]
+region = pygmt.info(
+    table=df[["sepal_width", "sepal_length"]],  # x and y columns
+    per_column=True,  # report output as a numpy array
+)
```

Comment:
Is it more Pythonic to use `min()` and `max()` functions？
```
region = [df.sepal_width.min(), df.sepal_width.max(), df.sepal_length.min(), df.sepal_length.max()]
```

Reply:
I prefer `pygmt.info` because it can actually round down and up a little to make the plot not so tight.

Reply:
That's a good point. I think we should use the `nearest_multiple` parameter to avoid plotting data points on the frame boundary.

Reply:
Agree with @weiji14. Added the ``spacing`` parameter to do that.